lib/libbento4/Core/Ap4Results.h

```diff
@@ -55,8 +55,6 @@ const int AP4_ERROR_INVALID_TRACK_TYPE              = -19;
 const int AP4_ERROR_INVALID_RTP_PACKET_EXTRA_DATA   = -20;
 const int AP4_ERROR_BUFFER_TOO_SMALL                = -21;
 const int AP4_ERROR_NOT_ENOUGH_DATA                 = -22;
-const int AP4_ERROR_NOT_ENOUGH_SPACE                = -23;
-const int AP4_ERROR_STREAM_CHANGED					= -24;
```

Comment:
@CastagnaIT 
You need to add "AP4_ERROR_STREAM_CHANGED" again as this is what is used here: https://github.com/xbmc/inputstream.adaptive/blob/26f21455a9427b5aa829297c5c7a5b9eb095af53/src/main.cpp#L226

Reply:
I think it's a bad idea to add an unknown constant to the upstream code,
Looking in the upstream code it seems that currently that there is no a specific error check in the ReadPartial method
with the exception of `AP4_ERROR_EOS`,
this suggests that something should be handled differently in our code, or as alternative use `AP4_ERROR_READ_FAILED` for both cases

Reply:
@glennguy take a look

Reply:
@CastagnaIT agreed, the constants for the most part aren't checked specifically, only whether a result is `AP4_SUCCESS` or `AP4_FAILED` (not `AP4_SUCCESS`). Maybe alter to  `AP4_FAILURE` or as you suggested?

Reply:
i think keep `AP4_ERROR_READ_FAILED` is appropriate others similar implementations of ReadPartial return this specific error when fails

Reply:
Sounds good

Reply:
add the change